mathics/core/rules.py

```diff
@@ -94,6 +94,8 @@ def __init__(self, pattern, replace, system=False):
     def do_replace(self, vars, options, evaluation):
         new = self.replace.replace_vars(vars)
         new.options = options
+        if len(options) > 0:  # need to trigger (re)evaluation?
```

Comment:
If I'm reading this correctly, this will happen whenever we do a replacement for a rule that contains an `OptionsPattern`. How come we're not applying the optimisation in this case?